code_generator_python/python_code_generator/templates/transform_single_file.py

```diff
@@ -22,58 +18,74 @@ def transform_single_file(file_path: str, output_path: Path, output_format: str)
     try:
         stime = time.time()
 
-        output = generated_transformer.run_query(file_path)
-
-        ttime = time.time()
-
-        if output_format == 'root-file':
+        # We first see if the function has the signature to directly write output
+        # If it doesn't, then we assume it's giving us back awkward array results
+        try:
```

Comment:
This is getting so ugly... could you please use the [`inspect.signature`](https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object) instead of this exception to make it a little more obvious what the intention is

Reply:
ok